crates/data_models/src/payments/payment_attempt.rs

```diff
@@ -308,6 +308,14 @@ pub enum PaymentAttemptUpdate {
         error_message: Option<Option<String>>,
         updated_by: String,
     },
+    BlocklistUpdate {
+        status: storage_enums::AttemptStatus,
+        error_code: Option<Option<String>>,
+        error_message: Option<Option<String>>,
+        updated_by: String,
+        connector: Option<Option<String>>,
+        merchant_connector_id: Option<Option<String>>,
```

Comment:
You don't need to put the `connector` and `merchant_connector_id` fields in the enum variant for blocklist update. Simply set the fields to `Some(None)` when converting the `PaymentAttemptUpdate` enum to `PaymentAttemptUpdateInternal`.